### PR TITLE
feat: add Rails 8 application CLAUDE.md example

### PR DESCRIPTION
**Pull request by rockwellwindsor**
Adds examples/rails-app-CLAUDE.md as a reference template for Rails 8 applications.

- Add examples/rails-app-CLAUDE.md: full-stack Rails 8 template covering Hotwire (Turbo + Stimulus), ViewComponent, the Solid stack (SolidQueue, SolidCache, SolidCable), service objects, query objects, and Pundit authorization
- Aligns with existing rules/ruby/ conventions (Rails Way first, SolidQueue for greenfield, Hotwire-preferred, Rails 8 generated authentication)
- Includes five Key Patterns code blocks: service object, skinny controller, query object, background job, RSpec test

## What Changed

Adds `examples/rails-app-CLAUDE.md`, a project-level CLAUDE.md template for Rails 8 applications. The file follows the structural pattern established by `django-api-CLAUDE.md`, `laravel-api-CLAUDE.md`, and `rust-api-CLAUDE.md`: Project Overview, Critical Rules, File Structure, Key Patterns, Environment Variables, Testing Strategy, ECC Workflow, and Git Workflow.

Coverage includes idiomatic Rails 8 conventions: Hotwire (Turbo + Stimulus) over JavaScript frameworks, the Solid stack (SolidQueue, SolidCache, SolidCable), the Rails 8 generated authentication system as default with Devise for complex flows, service objects namespaced by domain, query objects, ViewComponent for testable view logic, ActionCable conventions, and Kamal deployment setup.

## Why This Change

The `examples/` directory has CLAUDE.md templates for Django, Laravel, Rust, Next.js SaaS, Go, and HarmonyOS, but no Rails template. The README's "Ideas for Contributions" section explicitly lists Rails as a wanted framework contribution, and the existing `rules/ruby/` directory establishes Rails-aware coding conventions but no project-level example shows how those rules instantiate in a real Rails 8 application.

The template is positioned as a project-level companion to the existing `rules/ruby/` files. It aligns with conventions in `rules/ruby/patterns.md` (Hotwire-first, SolidQueue for greenfield, Rails 8 authentication generator as default), `rules/ruby/coding-style.md` (Rails Way first, service objects named by business operation), `rules/ruby/hooks.md` (RuboCop and Brakeman in CI), and `rules/ruby/security.md` (CSRF defaults, strong parameters, Rails credentials for secrets).

## Testing Done

Tested in a fresh Rails 8 personal project (a project and phase tracker) over an extended Claude Code session. Across seven distinct feature builds covering authentication setup, scope discipline on multi-concern features, controller naming conventions, dependency-aware work sequencing, N+1 prevention via eager loading, and Hotwire inline interactivity, Claude consistently cited either the project's CLAUDE.md or the ECC Ruby rules pack when applying conventions.

Specific behaviors observed:

- Authentication prompt resulted in Claude explicitly loading the five `rules/ecc/ruby/*.md` files into context and citing *"the Rails 8 authentication generator, which is the right tool for this (per the project's own CLAUDE.md)"* before implementing
- Vague "list of children under each parent" prompt resulted in Claude pre-committing to eager loading (*"eager load phases"*) before any code was written, with no performance cue in the prompt
- Interactive UI prompt resulted in Claude proposing Turbo Frame and Turbo Stream architecture (*"No full reload, no custom JS"*) without mention of any JavaScript framework alternative
- Multi-concern prompt resulted in correct scope separation (Project and Membership as distinct features) without prompting

Full test log with exact prompts and verbatim responses is available on request.

- [x] Manual testing completed
- [ ] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly (no JSON files added)
- [x] Shell scripts pass shellcheck (no shell scripts added)
- [ ] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation (this template is itself documentation)
- [ ] Added comments for complex logic (template is prose-and-code, not implementation)
- [ ] README updated (if needed) (no README change required for adding to examples/)

### CODE REVIEW

File: examples/rails-app-CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+      invoice = build_invoice
+
+      ApplicationRecord.transaction do
+        invoice.save!
+      end
+
+      send_notifications(invoice)
+      Result.new(success?: true, invoice: invoice, errors: nil)
+    rescue ActiveRecord::RecordInvalid => e
+      Result.new(success?: false, invoice: e.record, errors: e.record.errors)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply examples/rails-app-CLAUDE.md:167]: [0ab3e6f](https://github.com/affaan-m/ECC/pull/2258/commits/0ab3e6f3b2da543de1585e7889a2d8c15626c26b)
